flutter/lib/providers/superformula_provider.dart

```diff
@@ -0,0 +1,227 @@
+//Flutter packages
+import 'dart:convert';
+//Third party packages
+import 'package:dio/dio.dart';
+import 'package:flutter/material.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+//My Packages
+import '../utils.dart';
+import '../values.dart' as values;
+
+enum DOWNLOAD_STATE{
+  DOWNLOADING,
+  DOWNLOADED,
+  ERROR,
+}
+
+class SuperFormulaProvider with ChangeNotifier{
```

Comment:
**thought**: This class is doing too much. It should be broken up to more focused providers. Would be easier to test.